### ISSUE TITLE
 Missing multiplication with hyperedge weights W in HypergraphConv 

### ISSUE THREAD

**Issue by QbzQbz** [pyg-team/pytorch_geometric#6918]
### 🛠 Proposed Refactor

It seems missing multiplication with hyperedge weights W in HypergraphConv.

### Suggest a potential alternative/fix

        out = self.propagate(hyperedge_index, x=x, norm=B, hyperedge_weight=hyperedge_weight, alpha=alpha,
                             size=(num_nodes, num_edges), flag=True)# [num_edges, H, F]
        out = self.propagate(hyperedge_index.flip([0]), x=out, norm=D, hyperedge_weight=hyperedge_weight,
                             alpha=alpha, size=(num_edges, num_nodes), flag=False) # [num_nodes, H, F]

        if self.concat is True:
            out = out.view(-1, self.heads * self.out_channels)
        else:
            out = out.mean(dim=1)

        if self.bias is not None:
            out = out + self.bias

        return out

   
    def message(self, x_j: Tensor, norm_i: Tensor, hyperedge_weight_i, alpha: Tensor, flag) -> Tensor:
        H, F = self.heads, self.out_channels

        out = norm_i.view(-1, 1, 1) * x_j.view(-1, H, F) # [E,H,F]

        if flag == True:
            out = hyperedge_weight_i.view(-1, 1, 1) * out.view(-1, H, F)

        if alpha is not None:
            out = alpha.view(-1, self.heads, 1) * out

        return out

**Comment by QbzQbz** [comment]
According to Eq.3 and Eq.6 in the original paper _**Hypergraph Convolution and Hypergraph Attention**_, there should be a explicit W in the convolution.

**Comment by rusty1s** [comment]
Yeah, that is correct. We do this in our implementation as well, see [here](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/nn/conv/hypergraph_conv.py#L152), so edge weights are already captures through `norm`. Let me know if I am missing something.
